eng/pipelines/templates/steps/dotnet-diagnostics.yml

```
@@ -0,0 +1,13 @@
+parameters:
+- name: LogFilePath
+  type: string
+  default: $(Build.ArtifactStagingDirectory)/msbuild.binlog
+
+steps:
+- pwsh: |
+    if ('${{ eq(variables['System.Debug'], 'true') }}' -eq 'true') {
```

It still won't work if you're trying `${{ }}` because it gets parsed/rendered too early.

```suggestion
    if ('$(System.Debug)' -eq 'true') {
```